docs/development/design/rate-limiting.md

```diff
@@ -0,0 +1,35 @@
+# Used Envoy for Rate Limiting
+
+<!-- [Issue #600](https://github.com/redhat-et/issues/600) -->
+
+## Summary
+
+To protect the apiserver service from a misbehaving tenant, the REST APIs should be rate limited. This will allow us to better reason about how to scale the service to a large number of tenants as there will be limit to the amount of apiserver resources each tenant can consume.
```

Comment:
Can we clarify here that "tenant" is User and not Organization?

Reply:
+1